backends/advanced/src/advanced_omi_backend/services/transcription/deepgram.py

```
@@ -411,40 +16,42 @@ class DeepgramStreamConsumer:
-        self.api_key = api_key or os.getenv("DEEPGRAM_API_KEY")
-        if not self.api_key:
-            raise ValueError("DEEPGRAM_API_KEY is required")
-
-        # Initialize Deepgram provider
-        self.provider = DeepgramProvider(api_key=self.api_key)
+        # Get registry-driven transcription provider
+        self.provider = get_transcription_provider(mode="batch")
+        if not self.provider:
+            raise RuntimeError(
+                "Failed to load transcription provider. Ensure config.yml has a default 'stt' model configured."
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)